### PR TITLE
[4365] Send UK degree type to DQT

### PR DESCRIPTION
**Pull request by AbigailMcP**
### Context

https://trello.com/c/PgCx9CVV/4365-m-send-degree-type-to-dqt

DQT added support for accepting degree types. We need to send them when we make a TRN request and update a trainee.

### Changes proposed in this pull request

- Add a mapping between the UUIDs in the DfE Reference Data for degree
  types to the strings accepted by DQT
- Use this mapping to send degree type to DQT in the params for a TRN request
  and a trainee update.

Note: this is only intended to be temporary. There's a card on the TRA backlog to accept the UUIDs directly from the DfE Reference Data gem.

### Guidance to review

- Check out the DQT API specs to check that I'm sending the param correctly https://github.com/DFE-Digital/qualified-teachers-api/blob/main/docs/api-specs/v2.json#L1154
- Check out the spreadsheet on Trello to check the mapping

### Important business

- [ ] ~Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?`
- [ ] ~Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?~

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

**Comment by AbigailMcP** [comment]
[[M] Send degree type to DQT](https://trello.com/c/PgCx9CVV/4365-m-send-degree-type-to-dqt)

File: app/lib/dqt/params/trn_request.rb [hunk 1]
```diff
@@ -112,6 +112,7 @@ def qualification_params
           "subject" => subject_code,
           "class" => DEGREE_CLASSES[degree.grade],
           "date" => Date.parse("01-01-#{degree.graduation_year}").iso8601,
+          "heQualificationType" => CodeSets::DegreeTypes::MAPPING[degree.uk_degree_uuid],
```
Comment (markhallen) [inline app/lib/dqt/params/trn_request.rb:115]: Does this need a `.to_s` in case a nil is passed? Maybe we could add a test case for it.

**Comment by AbigailMcP** [comment]
There are still some discussions about whether we can add alternatives for the missing mappings https://ukgovernmentdfe.slack.com/archives/CGVNJ2Q2F/p1659696856266409

File: app/lib/dqt/params/trn_request.rb (hunk 1 above)
Comment (AbigailMcP) [reply app/lib/dqt/params/trn_request.rb]: Updated and test added!

**Review by markhallen — APPROVED** [review]
👍

**Pull request merged**